### PR TITLE
[skwasm] Fix skwasm clip coverage algorithm.

### PR DESCRIPTION
**Pull request by eyebrowsoffire**
This previously was basically assuming that two clip regions would be a union but instead it's actually an intersection. Also, for some reason I had forgotten to actually implement `boundingRect` for the rrect clip region. I changed this for a bit more clarity.

### CODE REVIEW

File: lib/web_ui/lib/src/engine/layers.dart [hunk 1]
```diff
@@ -543,12 +543,23 @@ class PlatformViewStyling {
...
+  ui.Rect get innerRect;
 
-  bool covers(ui.Rect rect);
+  /// The bounding rectangle of the clip region. All content outside of this
+  /// region is clipped.
+  ui.Rect get outerRect;
 
   ScenePath get toPath;
 
+  static bool rectCovers(ui.Rect covering, ui.Rect covered) {
```
Comment (kevmoo) [inline lib/web_ui/lib/src/engine/layers.dart:556]: https://api.flutter.dev/flutter/dart-ui/Rect/contains.html ?

Comment (kevmoo) [reply lib/web_ui/lib/src/engine/layers.dart:556]: er https://api.flutter.dev/flutter/dart-ui/Rect/overlaps.html ?

Comment (eyebrowsoffire) [reply lib/web_ui/lib/src/engine/layers.dart:556]: It is neither of those. `contains` acts on a single point. `overlaps` is that they have a non zero intersection, not that one completely covers the other (which is what we're checking).

Comment (kevmoo) [reply lib/web_ui/lib/src/engine/layers.dart:556]: ack. thanks for entertaining my silly

**Review by harryterkelsen — APPROVED** [review]
LGTM

**Pull request merged**